packages/editor-ui/src/components/TagsContainer.vue

```diff
@@ -1,133 +1,124 @@
-<script lang="ts">
-import { defineComponent, type ComponentInstance } from 'vue';
-
+<script lang="ts" setup>
+import { ref, computed, onMounted, onBeforeUnmount, nextTick } from 'vue';
+import type { ComponentInstance } from 'vue';
 import type { ITag } from '@/Interface';
 import IntersectionObserver from './IntersectionObserver.vue';
 import IntersectionObserved from './IntersectionObserved.vue';
-import { mapStores } from 'pinia';
-import { useTagsStore } from '@/stores/tags.store';
 import { createEventBus } from 'n8n-design-system/utils';
 import { debounce } from 'lodash-es';
 
-// random upper limit if none is set to minimize performance impact of observers
-const DEFAULT_MAX_TAGS_LIMIT = 20;
-
-interface TagEl extends ITag {
-	hidden?: boolean;
-	title?: string;
-	isCount?: boolean;
+// Define the interface for the props
+interface TagsContainerProps {
+	tagIds: string[];
+	tagsById: { [id: string]: ITag };
+	limit?: number;
+	clickable?: boolean;
+	responsive?: boolean;
+	hoverable?: boolean;
 }
 
-export default defineComponent({
-	name: 'TagsContainer',
-	components: { IntersectionObserver, IntersectionObserved },
-	props: {
-		tagIds: {
-			type: Array as () => string[],
-			required: true,
-		},
-		limit: {
-			type: Number,
-			default: DEFAULT_MAX_TAGS_LIMIT,
-		},
-		clickable: Boolean,
-		responsive: Boolean,
-		hoverable: Boolean,
-	},
-	emits: {
-		click: null,
-	},
-	data() {
-		return {
-			maxWidth: 320,
-			intersectionEventBus: createEventBus(),
-			visibility: {} as { [id: string]: boolean },
-			debouncedSetMaxWidth: () => {},
+// Define props using the interface and withDefaults
+const props = withDefaults(defineProps<TagsContainerProps>(), {
+	limit: 20, // DEFAULT_MAX_TAGS_LIMIT
+	clickable: false,
+	responsive: false,
+	hoverable: false,
+});
+
+// Define emits
+const emit = defineEmits<{
+	click: [tagId: string];
+}>();
+
+// Data
+const maxWidth = ref(320);
+const intersectionEventBus = createEventBus();
+const visibility = ref<{ [id: string]: boolean }>({});
+const tagsContainer = ref<ComponentInstance<typeof IntersectionObserver> | null>(null);
```

Comment:
Seeing this warning: `'any' overrides all other types in this union type.eslint[@typescript-eslint/no-redundant-type-constituents](https://typescript-eslint.io/rules/no-redundant-type-constituents)` which we could get rid of by removing the null type(and default value) and just leaving it undefined.
```suggestion
const tagsContainer = ref<ComponentInstance<typeof IntersectionObserver>>();
```